tests/src/Kernel/EntityMemoryCacheTest.php

```diff
@@ -0,0 +1,245 @@
+<?php
+
+/**
+ * Copyright 2018 Google Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License version 2 as published by the
+ * Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+namespace Drupal\Tests\apigee_edge\Kernel;
+
+use Apigee\Edge\Api\Management\Entity\CompanyApp;
+use Apigee\Edge\Api\Management\Entity\Developer;
+use Apigee\Edge\Api\Management\Entity\DeveloperApp;
+use Drupal\KernelTests\KernelTestBase;
+
+/**
+ * Apigee Edge entity memory cache tests.
+ *
+ * @group apigee_edge
+ * @group apigee_edge_kernel
+ */
+class EntityMemoryCacheTest extends KernelTestBase {
+
+  /**
+   * The entity cache implementation to test.
+   *
+   * @var \Drupal\apigee_edge\Entity\Controller\Cache\EntityCacheInterface
+   */
+  protected $entityCache;
+
+  /**
+   * The entity id cache implementation to test.
+   *
+   * @var \Drupal\apigee_edge\Entity\Controller\Cache\EntityIdCacheInterface
+   */
+  protected $entityIdCache;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'system',
+    'apigee_edge',
+    'key',
+  ];
+
+  /**
+   * Tests developer entity memory cache.
+   */
+  public function testDeveloperEntityMemoryCache() {
+    $this->entityCache = $this->container->get('apigee_edge.controller.cache.developer');
```

Comment:
Do not store these in class properties, rather create local variables and pass them as parameters to those functions that uses them, ex.: `assertEmptyCaches($entityCache, $entityIdCache)`